### PR TITLE
Add loading indication for audio

### LINKED ISSUE #1001: Add loading indications when audio results are being loaded
## Problem

We do not offer and feedback that audio is being loaded after the user clicks play or seeks an audio waveform. This makes the UI feel unresponsive.

## Description

We should add a loading animation.

IIRC @panchovm had a spinner in mind, potentially the new Gutenberg one from `@wordpress/components`. We'd have to port that from React to Vue.

We also need to research the a11y aspect of this and figure out the correct way to indicate this.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #1001 by @sarayourfriend 

## Description

Adds a loading indication to the button when audio is being loaded.

https://user-images.githubusercontent.com/24264157/172460095-b69a35eb-18f6-4784-869e-64c45b65ed0b.mp4

## Testing instructions
Do what I do in the video above, click around the audio results page, single results, mobile results, etc and make sure that the loading indicator appears and disappears when it's meant to. Be sure to test edge cases like multiple audios loading at once. Only the "about to be played" audio should ever indicate loading status.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by sarayourfriend** [comment]
@panchovm if you could take a look at this it'd be great. I kind of just guessed at the track color based on the screenshot you shared, but it's trivial to update so lmk if it needs to be changed. Same with the track width and size and everything.

File: src/components/VAudioTrack/VAudioTrack.vue [hunk 1]
```diff
@@ -131,11 +131,9 @@ export default defineComponent({
       // Preserve existing local audio if we plucked it from the global active audio
       if (!localAudio) localAudio = new Audio(props.audio.url)
 
-      localAudio.addEventListener('play', setPlaying)
-      localAudio.addEventListener('pause', setPaused)
-      localAudio.addEventListener('ended', setPlayed)
-      localAudio.addEventListener('timeupdate', setTimeWhenPaused)
-      localAudio.addEventListener('durationchange', setDuration)
+      Object.entries(eventMap).forEach(([name, fn]) =>
```
[the remark this answers is not in the document]
Comment (sarayourfriend) [reply src/components/VAudioTrack/VAudioTrack.vue:134]: Fixed with a cast in https://github.com/WordPress/openverse-frontend/pull/1492/commits/a78f5938a085cb73f2d6d6dd37f5b7288062a540

**Review by obulat — APPROVED** [review]
Works really well!

File: src/components/VAudioTrack/layouts/VGlobalLayout.vue [hunk 2]
```diff
@@ -38,6 +38,12 @@ export default defineComponent({
       required: true,
     },
   },
+  setup() {
+    const size = 'medium' as const
+    const layout = 'global' as const
+
+    return { size, layout }
+  },
```
Comment (krysal) [inline src/components/VAudioTrack/layouts/VGlobalLayout.vue:46]: Curious why to do this if it's not changing ❓

**Review by krysal — APPROVED** [review]
LGTM!

File: src/components/VAudioTrack/layouts/VGlobalLayout.vue (hunk 2 above)
Comment (sarayourfriend) [reply src/components/VAudioTrack/layouts/VGlobalLayout.vue:46]: Because there's no way to cast to `const` in the template :slightly_frowning_face: Vue JavaScript expressions in templates appear to be stuck on both an old version of JS syntax (no `??` for example) and also can't have TS in them. If we don't case `size` and `layout` here to `const` and pass them directly in the template as `size="medium"` they get passed as plain `string` types which doesn't match the type of the `size` prop (which is limited to the specific size strings). Same for `layout`.

It's a pain but at least with Vue it doesn't incur any significant runtime cost :crossed_fingers:

Comment (krysal) [reply src/components/VAudioTrack/layouts/VGlobalLayout.vue:46]: Ah, I see. Thanks for explanining!

**Comment by panchovm** [comment]
Jumping in late but coming to say that it looks awesome. Great work ✨
